diff --git a/vimeo-upload.js b/vimeo-upload.js
index 042e045..a268b4b 100644
--- a/vimeo-upload.js
+++ b/vimeo-upload.js
@@ -193,6 +193,7 @@
         xhr.open(this.httpMethod, this.url, true)
         xhr.setRequestHeader('Authorization', 'Bearer ' + this.token)
         xhr.setRequestHeader('Content-Type', 'application/json')
+        xhr.setRequestHeader('Accept', 'application/vnd.vimeo.*+json;version=3.2')
 
         xhr.onload = function(e) {
             // get vimeo upload  url, user (for available quote), ticket id and complete url
