diff --git a/golive_checklist.info b/golive_checklist.info
deleted file mode 100644
index f3468f6..0000000
--- a/golive_checklist.info
+++ /dev/null
@@ -1,6 +0,0 @@
-name = GOLIVE Checklist
-description = A checklist for taking a Drupal site live.
-core = 7.x
-dependencies[] = checklistapi
-package = GOLIVE
-configure = admin/config/search/golive-checklist
diff --git a/golive_checklist.info.yml b/golive_checklist.info.yml
new file mode 100644
index 0000000..8abfdd7
--- /dev/null
+++ b/golive_checklist.info.yml
@@ -0,0 +1,8 @@
+name: GOLIVE Checklist
+description: A checklist for taking a Drupal site live.
+core: 8.x
+package: GOLIVE
+dependencies:
+  - checklistapi
+type: module
+configure: checklistapi.checklists.golive_checklist
diff --git a/golive_checklist.install b/golive_checklist.install
index 6e0b7c8..ecc3892 100644
--- a/golive_checklist.install
+++ b/golive_checklist.install
@@ -9,5 +9,7 @@
  * Implements hook_uninstall().
  */
 function golive_checklist_uninstall() {
-  variable_del('checklistapi_checklist_golive_checklist');
+  \Drupal::configFactory()
+    ->getEditable("checklistapi.progress.golive_checklist")
+    ->delete();
 }
diff --git a/golive_checklist.js b/golive_checklist.js
index fc2f20e..0f38635 100644
--- a/golive_checklist.js
+++ b/golive_checklist.js
@@ -1,4 +1,4 @@
-(function ($) {
+(function ($, Drupal, drupalSettings) {
 
   "use strict";
   Drupal.behaviors.seo_checklist = {
@@ -20,4 +20,4 @@
     }
   };
 
-})(jQuery);
+})(jQuery, Drupal, drupalSettings);
diff --git a/golive_checklist.libraries.yml b/golive_checklist.libraries.yml
new file mode 100644
index 0000000..e2d318c
--- /dev/null
+++ b/golive_checklist.libraries.yml
@@ -0,0 +1,7 @@
+golive_checklist:
+  version: 1.x
+  js:
+    golive_checklist.js: {}
+  css:
+    theme:
+      golive_checklist.css: {}
diff --git a/golive_checklist.links.menu.yml b/golive_checklist.links.menu.yml
new file mode 100644
index 0000000..11b0483
--- /dev/null
+++ b/golive_checklist.links.menu.yml
@@ -0,0 +1,5 @@
+golive_checklist.checklist:
+  title: Golive Checklist
+  description: Golive Checklist.
+  route_name: checklistapi.checklists.golive_checklist
+  parent: system.admin_config_development
diff --git a/golive_checklist.module b/golive_checklist.module
index e3d7779..4cc4646 100644
--- a/golive_checklist.module
+++ b/golive_checklist.module
@@ -5,6 +5,8 @@
  * Do important actions on the website before going live.
  */
 
+use Drupal\Core\Url;
+
 /**
  * Checks whether the Urlaub im GLas footer link is enabled.
  *
@@ -14,8 +16,9 @@
 function _golive_checklist_urlaubimglas_link_is_enabled() {
   $link_is_enabled = &drupal_static(__FUNCTION__);
   if (!isset($link_is_enabled)) {
-    $saved_progress = variable_get('checklistapi_checklist_golive_checklist', array());
-    $link_is_enabled = !empty($saved_progress['link_to_urlaubimglas']);
+    $config = \Drupal::config('checklistapi.progress.golive_checklist');
+    $settings = $config->get();
+    $link_is_enabled = !empty($settings['link_to_urlaubimglas']);
   }
   return $link_is_enabled;
 }
@@ -25,8 +28,7 @@ function _golive_checklist_urlaubimglas_link_is_enabled() {
  */
 function golive_checklist_form_checklistapi_checklist_form_alter(&$form, &$form_state, $form_id) {
   if ($form['#checklist']->id == 'golive_checklist') {
-    $form['checklistapi']['#attached']['js'][] = drupal_get_path('module', 'golive_checklist') . '/golive_checklist.js';
-    $form['checklistapi']['#attached']['css'][] = drupal_get_path('module', 'golive_checklist') . '/golive_checklist.css';
+    $form['#attached']['library'][] = 'golive_checklist/golive_checklist';
   }
 }
 
@@ -34,7 +36,21 @@ function golive_checklist_form_checklistapi_checklist_form_alter(&$form, &$form_
  * Implements hook_checklistapi_checklist_info().
  */
 function golive_checklist_checklistapi_checklist_info() {
-  $modules_page = array('#text' => t('Enable'), '#path' => 'admin/modules');
+  $modules_url = Url::fromRoute('system.modules_list', array(), array('absolute' => TRUE));
+  $modules_page = array(
+    '#text' => t('Enable'),
+    '#url' => $modules_url,
+  );
+  $urlaubimglas = array(
+    '#theme' => 'image',
+    '#uri' => drupal_get_path('module', 'golive_checklist') . '/urlaubimglas-logo.png',
+    '#alt' => t('Urlaub im Glas'),
+    '#title' => t('Urlaub im Glas'),
+  );
+  $intro_tab = array(
+    '#theme' => 'golive_checklist_intro_tab',
+    '#webozy_logo' => \Drupal::service('renderer')->render($urlaubimglas),
+  );
 
   // TODO:
   // Configure and enable Mollom or spam filtering.
@@ -43,29 +59,29 @@ function golive_checklist_checklistapi_checklist_info() {
   // Change the email address for the site mail on the Site Info admin page.
   // Run a link checker if it's an old/migrated site.
   // Update any necessary API keys if they were used on the dev/stage sites
-  //     (Google Analytics, Mollom, GMaps, Flickr, etc.)
+  // (Google Analytics, Mollom, GMaps, Flickr, etc.)
   // Install Nagios (status monitoring) module and file an internal ticket to
-  //     have its monitoring set up.
+  // have its monitoring set up.
   // Check the status report: admin/reports/status
   // Optimize all images with tools such as optipng, jpegoptim, Smush.it, ...
   // Check if one of the domain redirect options is enabled in the .htaccess
-  //     file. Either redirect URLs without www. or vice versa.
+  // file. Either redirect URLs without www. or vice versa.
   // On the Permissions page, verify what permissions anonymous and
-  //     authenticated users have been given
+  // authenticated users have been given
   // On the User Settings page, verify that account creation settings are as
-  //     you intend (can users create accounts? do they need approval?)
+  // you intend (can users create accounts? do they need approval?)
   // https://www.drupal.org/project/ultimate_cron
   // http://www.microserveltd.co.uk/blogs/drupal-hosting/high-performance-drupal-part-1-give-your-site-boost
   // http://www.microserveltd.co.uk/blogs/drupal-development/high-performance-drupal-part-2-lightning-fast-code
   // Remove test content, such as "lorum ipsum" text, dummy users, or content
-  //     generated by the Devel module.
+  // generated by the Devel module.
   // Review Google Webmaster Tools for index issues
   // https://www.drupal.org/project/mysqlreport
   // https://www.drupal.org/project/shadow
   // BUMP UP YOUR SERVER MEMORY LIMIT If you check your site's status report,
-  //     you will see what your PHP memory limit is set to. More than likely
-  //     this is set to 64M. This is the amount of memory that your server
-  //     allocates to running the PHP scripts that power your site.
+  // you will see what your PHP memory limit is set to. More than likely
+  // this is set to 64M. This is the amount of memory that your server
+  // allocates to running the PHP scripts that power your site.
   // enable clean URLs
   // Path module
   // Pathauto
@@ -87,13 +103,7 @@ function golive_checklist_checklistapi_checklist_info() {
     // Introduction.
     'introduction' => array(
       '#title' => t('Introduction'),
-      '#description' => theme('golive_checklist_intro_tab', array(
-        'urlaubimglas_logo' => theme('image', array(
-          'path' => drupal_get_path('module', 'golive_checklist') . '/urlaubimglas-logo.png',
-          'alt' => t('Urlaub im Glas'),
-          'title' => t('Urlaub im Glas'),
-        )),
-      )),
+      '#description' => \Drupal::service('renderer')->render($intro_tab),
     ),
     // Configuration.
     'configuration' => array(
@@ -108,28 +118,28 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('[Optional] Disable and uninstall all not used core and non-core modules.'),
         'modules_page' => array(
           '#text' => t('Disable'),
-          '#path' => 'admin/modules'
+          '#url' => $modules_url,
         ),
       ),
       'disable_devel' => array(
         '#title' => t('[Optional] Disable and uninstall all development modules like "devel".'),
         'modules_page' => array(
           '#text' => t('Disable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
       ),
       'disable_ui' => array(
         '#title' => t('[Optional] Disable and uninstall all UI modules like "views UI".'),
         'modules_page' => array(
           '#text' => t('Disable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
       ),
       'disable_overlay' => array(
         '#title' => t('[Optional] Disable and uninstall overlay module.'),
         'modules_page' => array(
           '#text' => t('Disable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
       ),
       'bootstrap_optimizer' => array(
@@ -137,11 +147,11 @@ function golive_checklist_checklistapi_checklist_info() {
         '#description' => t('Module is searching for all modules that were deleted unproperly. Bootstrap optimizer removes this modules from {system} table. It allows Drupal to skip heavy modules/themes directory scan during every page load.'),
         'modules_page' => array(
           '#text' => t('Enable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
         'l10n_update' => array(
           '#text' => t('Bootstrap optimizer'),
-          '#path' => 'https://www.drupal.org/project/bootstrap_optimizer',
+          '#url' => Url::fromUri('https://www.drupal.org/project/bootstrap_optimizer'),
         ),
       ),
       'missing_module' => array(
@@ -149,11 +159,11 @@ function golive_checklist_checklistapi_checklist_info() {
         '#description' => t('This module lists modules that are activated in your database but missing from your file system.'),
         'modules_page' => array(
           '#text' => t('Enable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
         'l10n_update' => array(
           '#text' => t('Missing module'),
-          '#path' => 'https://www.drupal.org/project/missing_module',
+          '#url' => Url::fromUri('https://www.drupal.org/project/missing_module'),
         ),
       ),
       'clean_missing_module' => array(
@@ -161,11 +171,11 @@ function golive_checklist_checklistapi_checklist_info() {
         '#description' => t('Provides a Drush command that will find missing modules and disable them in the database.'),
         'modules_page' => array(
           '#text' => t('Enable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
         'l10n_update' => array(
           '#text' => t('Clean missing module'),
-          '#path' => 'https://www.drupal.org/project/clean_missing_modules',
+          '#url' => Url::fromUri('https://www.drupal.org/project/clean_missing_modules'),
         ),
       ),
     ),
@@ -177,57 +187,57 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('[Optional] Set up languages.'),
         'language_page' => array(
           '#text' => t('Languages'),
-          '#path' => 'admin/config/regional/language',
+          '#url' => \Drupal::moduleHandler()->moduleExists('language') ? Url::fromRoute('entity.configurable_language.collection', array(), array('absolute' => TRUE)) : $modules_url,
         ),
       ),
       'default_language' => array(
         '#title' => t('[Optional] Set default language.'),
         'language_page' => array(
           '#text' => t('Languages'),
-          '#path' => 'admin/config/regional/language',
+          '#url' => \Drupal::moduleHandler()->moduleExists('language') ? Url::fromRoute('entity.configurable_language.collection', array(), array('absolute' => TRUE)) : $modules_url,
         ),
       ),
       'detection' => array(
         '#title' => t('[Optional] Configured detection and selection.'),
         'detection_page' => array(
           '#text' => t('Detection and selection'),
-          '#path' => 'admin/config/regional/language/configure',
+          '#url' => \Drupal::moduleHandler()->moduleExists('language') ? Url::fromRoute('language.negotiation', array(), array('absolute' => TRUE)) : $modules_url,
         ),
       ),
       'l10n_update' => array(
-        '#default_value' => module_exists('l10n_update'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('l10n_update'),
         '#title' => t('[Optional] Installed l10n_update module.'),
         'modules_page' => array(
           '#text' => t('Enable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
         'l10n_update' => array(
           '#text' => t('Localization update'),
-          '#path' => 'https://www.drupal.org/project/l10n_update',
+          '#url' => Url::fromUri('https://www.drupal.org/project/l10n_update'),
         ),
       ),
       'content_translation' => array(
-        '#default_value' => module_exists('translation'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('translation'),
         '#title' => t('[Optional] Installed content translation module.'),
         'modules_page' => array(
           '#text' => t('Enable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
         'content_translation' => array(
           '#text' => t('Content translation'),
-          '#path' => 'https://www.drupal.org/project/translation',
+          '#url' => Url::fromUri('https://www.drupal.org/project/translation'),
         ),
       ),
       'entity_translation' => array(
-        '#default_value' => module_exists('entity_translation'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('entity_translation'),
         '#title' => t('[Optional] Installed entity translation module.'),
         'modules_page' => array(
           '#text' => t('Enable'),
-          '#path' => 'admin/modules',
+          '#url' => $modules_url,
         ),
         'entity_translation' => array(
           '#text' => t('Entity translation'),
-          '#path' => 'https://www.drupal.org/project/entity_translation',
+          '#url' => Url::fromUri('https://www.drupal.org/project/entity_translation'),
         ),
       ),
     ),
@@ -239,11 +249,11 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => 'Set session time in settings.php',
       ),
       'autologout' => array(
-        '#default_value' => module_exists('autologout'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('autologout'),
         '#title' => t('Enable autologout'),
         'autologout' => array(
           '#text' => t('Autologout'),
-          '#path' => 'https://www.drupal.org/project/autologout',
+          '#url' => Url::fromUri('https://www.drupal.org/project/autologout'),
         ),
       ),
       'php_errors' => array(
@@ -251,53 +261,53 @@ function golive_checklist_checklistapi_checklist_info() {
       ),
       'security_review' => array(
         '#title' => t('[Optional] Install Security Review module.'),
-        '#default_value' => module_exists('security_review'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('security_review'),
         'project_page' => array(
           '#text' => t('Download'),
-          '#path' => 'http://drupal.org/project/security_review',
+          '#url' => Url::fromUri('http://drupal.org/project/security_review'),
         ),
         'modules_page' => $modules_page,
       ),
       'captcha' => array(
         '#title' => t('Enaple Captcha'),
-        '#default_value' => module_exists('captcha'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('captcha'),
         'captcha' => array(
           '#text' => t('Captcha'),
-          '#path' => 'https://www.drupal.org/project/captcha',
+          '#url' => Url::fromUri('https://www.drupal.org/project/captcha'),
         ),
         'modules_page' => $modules_page,
       ),
       'riddler' => array(
         '#title' => t('Riddler'),
-        '#default_value' => module_exists('riddler'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('riddler'),
         'riddler' => array(
           '#text' => t('Riddler'),
-          '#path' => 'https://www.drupal.org/project/riddler',
+          '#url' => Url::fromUri('https://www.drupal.org/project/riddler'),
         ),
       ),
       'admin_password' => array(
         '#title' => t('Select secure admin password'),
         'user' => array(
           '#text' => t('User admin'),
-          '#path' => 'users/admin',
+          '#url' => Url::fromRoute('entity.user.canonical', array('user' => 1), array('absolute' => TRUE)),
         ),
       ),
       'updates' => array(
         '#title' => t('Check for updates.'),
         'updates_pages' => array(
           '#text' => t('List'),
-          '#path' => 'admin/reports/updates',
+          '#url' => \Drupal::moduleHandler()->moduleExists('update') ? Url::fromRoute('update.status', array(), array('absolute' => TRUE)) : Url::fromRoute('system.modules_list', array(), array('absolute' => TRUE)),
         ),
         'updates_check' => array(
           '#text' => t('Check'),
-          '#path' => 'admin/reports/updates/update',
+          '#url' => \Drupal::moduleHandler()->moduleExists('update') ? Url::fromRoute('update.report_update', array(), array('absolute' => TRUE)) : Url::fromRoute('system.modules_list', array(), array('absolute' => TRUE)),
         ),
       ),
       'updates_setup' => array(
         '#title' => t('Setup E-mail notifications about updates.'),
         'updates_setup' => array(
           '#text' => t('Setup'),
-          '#path' => 'admin/reports/updates/settings',
+          '#url' => \Drupal::moduleHandler()->moduleExists('update') ? Url::fromRoute('update.settings', array(), array('absolute' => TRUE)) : Url::fromRoute('system.modules_list', array(), array('absolute' => TRUE)),
         ),
       ),
       'session_lifetime' => array(
@@ -307,7 +317,7 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('Securing file permissions.'),
         'file_permissions' => array(
           '#text' => t('Setup'),
-          '#path' => 'https://www.drupal.org/node/244924',
+          '#url' => Url::fromUri('https://www.drupal.org/node/244924'),
         ),
       ),
       'php_filter' => array(
@@ -321,7 +331,7 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('Deactivate error messages.'),
         'error_page' => array(
           '#text' => t('Errors'),
-          '#path' => 'admin/config/development/logging',
+          '#url' => Url::fromRoute('system.logging_settings', array(), array('absolute' => TRUE)),
         ),
       ),
 
@@ -342,42 +352,42 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('Enable caching.'),
         'performance_page' => array(
           '#text' => t('Caching'),
-          '#path' => 'admin/config/development/performance',
+          '#url' => Url::fromRoute('system.performance_settings', array(), array('absolute' => TRUE)),
         ),
       ),
       'apc' => array(
         '#title' => t('Use apc.'),
-        '#default_value' => module_exists('apc'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('apc'),
         'apc_page' => array(
           '#text' => t('Download'),
-          '#path' => 'http://drupal.org/project/apc',
+          '#url' => Url::fromUri('http://drupal.org/project/apc'),
         ),
         'modules_page' => $modules_page,
       ),
       'memcache' => array(
         '#title' => t('Use memcache.'),
-        '#default_value' => module_exists('memcache'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('memcache'),
         'memcache_page' => array(
           '#text' => t('Download'),
-          '#path' => 'http://drupal.org/project/memcache',
+          '#url' => Url::fromUri('http://drupal.org/project/memcache'),
         ),
         'modules_page' => $modules_page,
       ),
       'memcache_storage' => array(
         '#title' => t('Use memcache_storage instead.'),
-        '#default_value' => module_exists('memcache_storage'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('memcache_storage'),
         'memcache_page' => array(
           '#text' => t('Download'),
-          '#path' => 'http://drupal.org/project/memcache_storage',
+          '#url' => Url::fromUri('http://drupal.org/project/memcache_storage'),
         ),
         'modules_page' => $modules_page,
       ),
       'boost' => array(
         '#title' => t('Use boost.'),
-        '#default_value' => module_exists('boost'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('boost'),
         'memcache_page' => array(
           '#text' => t('Download'),
-          '#path' => 'http://drupal.org/project/boost',
+          '#url' => Url::fromUri('http://drupal.org/project/boost'),
         ),
         'modules_page' => $modules_page,
       ),
@@ -385,22 +395,22 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('Increase Memory Limit.'),
         'memory_page' => array(
           '#text' => t('How to'),
-          '#path' => 'http://drupal.org/node/207036',
+          '#url' => Url::fromUri('http://drupal.org/node/207036'),
         ),
       ),
       'Aggregation' => array(
         '#title' => t('Turn on aggregation for CSS and JavaScript files.'),
         'performance_page' => array(
           '#text' => t('Caching'),
-          '#path' => 'admin/config/development/performance',
+          '#url' => Url::fromRoute('system.performance_settings', array(), array('absolute' => TRUE)),
         ),
       ),
       'performance' => array(
         '#title' => t('Monitor your performance.'),
-        '#default_value' => module_exists('performance'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('performance'),
         'performance_page' => array(
           '#text' => t('Download'),
-          '#path' => 'https://www.drupal.org/project/performance',
+          '#url' => Url::fromUri('https://www.drupal.org/project/performance'),
         ),
         'modules_page' => $modules_page,
       ),
@@ -413,7 +423,7 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('Schedule on cron.'),
         'cron_page' => array(
           '#text' => t('Cron'),
-          '#path' => 'admin/config/system/cron',
+          '#url' => Url::fromRoute('system.cron_settings', array(), array('absolute' => TRUE)),
         ),
       ),
       'cronjob' => array(
@@ -421,19 +431,19 @@ function golive_checklist_checklistapi_checklist_info() {
       ),
       'elysia' => array(
         '#title' => t('Use Elysia Cron.'),
-        '#default_value' => module_exists('elysia_cron'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('elysia_cron'),
         'elysia_page' => array(
           '#text' => t('Download'),
-          '#path' => 'https://www.drupal.org/project/elysia_cron',
+          '#url' => Url::fromUri('https://www.drupal.org/project/elysia_cron'),
         ),
         'modules_page' => $modules_page,
       ),
       'ultimate_cron' => array(
         '#title' => t('Use Ultimate Cron.'),
-        '#default_value' => module_exists('ultimate_cron'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('ultimate_cron'),
         'ultimate_page' => array(
           '#text' => t('Download'),
-          '#path' => 'https://www.drupal.org/project/ultimate_cron',
+          '#url' => Url::fromUri('https://www.drupal.org/project/ultimate_cron'),
         ),
         'modules_page' => $modules_page,
       ),
@@ -446,41 +456,41 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('Disable unused views pages.'),
         'views_page' => array(
           '#text' => t('Views'),
-          '#path' => 'admin/structure/views',
+          '#url' => Url::fromRoute('entity.view.collection', array(), array('absolute' => TRUE)),
         ),
       ),
       'node' => array(
         '#title' => t('Disable www.example.com/node.'),
-        '#default_value' => module_exists('bs_nice_frontpage'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('bs_nice_frontpage'),
         'nice_frontpage' => array(
           '#text' => t('Nice Frontpage'),
-          '#path' => 'http://drupal.org/project/bs_nice_frontpage',
+          '#url' => Url::fromUri('http://drupal.org/project/bs_nice_frontpage'),
         ),
         'custom_module' => array(
           '#text' => t('Custom Module'),
-          '#path' => 'http://dominiquedecooman.com/blog/how-remove-drupal-default-node-page-aka-default-front-page-view',
+          '#url' => Url::fromUri('http://dominiquedecooman.com/blog/how-remove-drupal-default-node-page-aka-default-front-page-view'),
         ),
       ),
       'path' => array(
         '#title' => t('Enable path.'),
-        '#default_value' => module_exists('path'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('path'),
         'modules_page' => $modules_page,
       ),
       'pathauto' => array(
         '#title' => t('Enable pathauto.'),
-        '#default_value' => module_exists('pathauto'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('pathauto'),
         'pathauto' => array(
           '#text' => t('Pathauto'),
-          '#path' => 'http://drupal.org/project/pathauto',
+          '#url' => Url::fromUri('http://drupal.org/project/pathauto'),
         ),
         'modules_page' => $modules_page,
       ),
       'menu_trail_by_path' => array(
         '#title' => t('Enable menu trail by path.'),
-        '#default_value' => module_exists('menu_trail_by_path'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('menu_trail_by_path'),
         'menu_trail_by_path' => array(
           '#text' => t('Menu Trail By Path'),
-          '#path' => 'http://drupal.org/project/menu_trail_by_path',
+          '#url' => Url::fromUri('http://drupal.org/project/menu_trail_by_path'),
         ),
         'modules_page' => $modules_page,
       ),
@@ -495,10 +505,10 @@ function golive_checklist_checklistapi_checklist_info() {
       ),
       'backup_and_migrate' => array(
         '#title' => t('Backup and Migrate'),
-        '#default_value' => module_exists('backup_migrate'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('backup_migrate'),
         'backup_and_migrate' => array(
           '#text' => 'Backup and Migrate',
-          '#path' => 'http://drupal.org/project/backup_migrate',
+          '#url' => Url::fromUri('http://drupal.org/project/backup_migrate'),
         ),
         'modules_page' => $modules_page,
       ),
@@ -546,15 +556,15 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('Site name etc.'),
         'title' => array(
           '#text' => t('Name'),
-          '#path' => 'admin/config/system/site-information',
+          '#url' => Url::fromRoute('system.site_information_settings', array(), array('absolute' => TRUE)),
         ),
       ),
       'stmp' => array(
         '#title' => t('SMTP'),
-        '#default_value' => module_exists('smtp'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('smtp'),
         'smtp' => array(
           '#text' => t('SMTP'),
-          '#path' => 'https://www.drupal.org/project/smtp',
+          '#url' => Url::fromUri('https://www.drupal.org/project/smtp'),
         ),
         'modules_page' => $modules_page,
       ),
@@ -562,7 +572,7 @@ function golive_checklist_checklistapi_checklist_info() {
         '#title' => t('ensure that the sender email address and SMTP settings are correct'),
         'smtp_settings' => array(
           '#text' => t('SMTP settings'),
-          '#path' => 'admin/config/system/smtp',
+          '#url' => \Drupal::moduleHandler()->moduleExists('smtp') ? Url::fromRoute('smtp.config', array(), array('absolute' => TRUE)) : $modules_url,
         ),
       ),
     ),
@@ -575,23 +585,23 @@ function golive_checklist_checklistapi_checklist_info() {
         'modules_page' => $modules_page,
         'setup' => array(
           '#text' => t('Settings'),
-          '#path' => 'admin/config/system/statistics',
+          '#url' => \Drupal::moduleHandler()->moduleExists('smtp') ? Url::fromRoute('statistics.settings', array(), array('absolute' => TRUE)) : $modules_url,
         ),
       ),
       'g_analytics' => array(
         '#title' => t('Enable Google Analytics'),
-        '#default_value' => module_exists('google_analytics'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('google_analytics'),
         'analytics' => array(
           '#text' => 'Analyitcs',
-          '#path' => 'https://www.drupal.org/project/google_analytics',
+          '#url' => Url::fromUri('https://www.drupal.org/project/google_analytics'),
         ),
       ),
       'piwik' => array(
         '#title' => t('Enable Piwik Web Analytics'),
-        '#default_value' => module_exists('piwik'),
+        '#default_value' => \Drupal::moduleHandler()->moduleExists('piwik'),
         'piwik' => array(
           '#text' => 'Piwik Web Analytics',
-          '#path' => 'https://www.drupal.org/project/piwik',
+          '#url' => Url::fromUri('https://www.drupal.org/project/piwik'),
         ),
       ),
     ),
@@ -620,13 +630,15 @@ function golive_checklist_checklistapi_checklist_info() {
 }
 
 /**
- * Implements hook_page_alter().
+ * Implements hook_page_bottom().
  */
-function golive_checklist_page_alter(&$page) {
+function golive_checklist_page_top(array &$page_bottom) {
   if (_golive_checklist_urlaubimglas_link_is_enabled()) {
-    $page['page_bottom']['golive_checklist_urlaubimglas_link'] = array(
-      '#type' => 'markup',
+    $page_bottom['golive_checklist_urlaubimglas_link'] = array(
       '#markup' => '<div id="golive-checklist-urlaubimglas-link"><a href="http://www.urlaubimglas.de">Urlaub im Glas Fruchtaufstriche</a></div>',
+      '#attached' => array(
+        'library' => 'golive_checklist/golive_checklist',
+      ),
     );
   }
 }
@@ -637,7 +649,6 @@ function golive_checklist_page_alter(&$page) {
 function golive_checklist_theme($existing, $type, $theme, $path) {
   return array(
     'golive_checklist_intro_tab' => array(
-      'template' => 'golive_checklist_intro_tab',
       'variables' => array(
         'urlaubimglas_logo' => t('Urlaub im Glas'),
       ),
diff --git a/golive_checklist.routing.yml b/golive_checklist.routing.yml
new file mode 100644
index 0000000..7b8a6a2
--- /dev/null
+++ b/golive_checklist.routing.yml
@@ -0,0 +1,9 @@
+checklistapi.checklists.golive_checklist:
+  path: '/admin/config/development/golive-checklist'
+  defaults:
+    _title: 'Golive Checklist'
+    _form: '\Drupal\checklistapi\Form\ChecklistapiChecklistForm'
+    checklist_id: golive_checklist
+    op: any
+  requirements:
+    _checklistapi_access: 'TRUE'
diff --git a/golive_checklist_intro_tab.tpl.php b/golive_checklist_intro_tab.tpl.php
deleted file mode 100644
index 4069fd3..0000000
--- a/golive_checklist_intro_tab.tpl.php
+++ /dev/null
@@ -1,50 +0,0 @@
-<?php
-
-/**
- * @file
- * Default theme implementation for the GOLIVE Checklist "Intro" tab.
- *
- * Available variables:
- * - $urlaubimglas_logo: An HTML image tag for the Urlaub im Glas logo.
- *
- * @see template_preprocess()
- * @see template_preprocess_seo_checklist_intro_tab()
- * @see template_process()
- */
-?>
-<h3>How to use the Drupal SEO Checklist</h3>
-<p>
-  This checklist is supposed to help you in your process of going live with your website.
-  If you are in charge of launching a Drupal website, then this checklist is for you.
-  My intention with this list is to have a collection of things you should consider.
-  Depending on your site and your setup not all steps might be required.
-  So don't use this list as a have-to-do but rather as a could-help list.
-</p>
-
-<h4>References</h4>
-
-<ul>
-  <li><a href="http://www.studiosdigital.at/blog/drupal-going-live-checklist">http://www.studiosdigital.at/blog/drupal-going-live-checklist</a></li>
-  <li><a href="https://www.drupal.org/node/672424">https://www.drupal.org/node/672424</a></li>
-  <li><a href="http://www.microserveltd.co.uk/blogs/drupal-development/going-live-drupal-checklist">http://www.microserveltd.co.uk/blogs/drupal-development/going-live-drupal-checklist</a></li>
-  <li><a href="https://www.drupal.org/node/1076460">https://www.drupal.org/node/1076460</a></li>
-  <li><a href="https://www.urbaninsight.com/2011/06/27/14-must-do-items-your-drupal-post-launch-checklist">https://www.urbaninsight.com/2011/06/27/14-must-do-items-your-drupal-post-launch-checklist</a></li>
-  <li><a href="http://www.onextrapixel.com/2011/10/06/from-design-to-live-15-must-do-steps-after-installing-drupal/">http://www.onextrapixel.com/2011/10/06/from-design-to-live-15-must-do-steps-after-installing-drupal/</a></li>
-</ul>
-<br/>
-<ul>
-  <li><a href="http://www.devsaran.com/blog/6-drupal-seo-modules-improve-your-drupal-site-rankings">http://www.devsaran.com/blog/6-drupal-seo-modules-improve-your-drupal-site-rankings</a></li>
-  <li><a href="http://poplarware.com/articles/drupal_cheat_sheet#checklist">http://poplarware.com/articles/drupal_cheat_sheet#checklist</a>
-  </li>
-</ul>
-
-<h4>Credits</h4>
-<p>The Drupal Golive Checklist was created by <a href="http://dgo.to/@bechtold">Oskar Bechtold</a>, the CEO of <a
-    href="http://www.urlaubimglas.de">Urlaub im Glas</a> and long-time Drupal community member. Development was paid
-  for exclusively by Urlaub im Glas. Special thanks to <a href="http://dgo.to/@traviscarden">Travis Carden</a> who
-  created the <a href="http://dgo.to/checklistapi">Checklist API</a> and to <a href="http://dgo.to/@ben-finklea">Ben
-    Finklea</a> who created <a href="http://dgo.to/seo_checklist">Checklist API</a> which was mainly used as an
-  inspiration of the technical implementation of GOLIVE Checklist. </p>
-<p id="golive-checklist-intro-urlaubimglas">
-  <a href="http://www.urlaubimglas.de"><?php print $urlaubimglas_logo; ?></a>
-</p>
diff --git a/templates/golive-checklist-intro-tab.html.twig b/templates/golive-checklist-intro-tab.html.twig
new file mode 100644
index 0000000..5d79b85
--- /dev/null
+++ b/templates/golive-checklist-intro-tab.html.twig
@@ -0,0 +1,36 @@
+<h3>How to use the Drupal SEO Checklist</h3>
+<p>
+  This checklist is supposed to help you in your process of going live with your website.
+  If you are in charge of launching a Drupal website, then this checklist is for you.
+  My intention with this list is to have a collection of things you should consider.
+  Depending on your site and your setup not all steps might be required.
+  So don't use this list as a have-to-do but rather as a could-help list.
+</p>
+
+<h4>References</h4>
+
+<ul>
+  <li><a href="http://www.studiosdigital.at/blog/drupal-going-live-checklist">http://www.studiosdigital.at/blog/drupal-going-live-checklist</a></li>
+  <li><a href="https://www.drupal.org/node/672424">https://www.drupal.org/node/672424</a></li>
+  <li><a href="http://www.microserveltd.co.uk/blogs/drupal-development/going-live-drupal-checklist">http://www.microserveltd.co.uk/blogs/drupal-development/going-live-drupal-checklist</a></li>
+  <li><a href="https://www.drupal.org/node/1076460">https://www.drupal.org/node/1076460</a></li>
+  <li><a href="https://www.urbaninsight.com/2011/06/27/14-must-do-items-your-drupal-post-launch-checklist">https://www.urbaninsight.com/2011/06/27/14-must-do-items-your-drupal-post-launch-checklist</a></li>
+  <li><a href="http://www.onextrapixel.com/2011/10/06/from-design-to-live-15-must-do-steps-after-installing-drupal/">http://www.onextrapixel.com/2011/10/06/from-design-to-live-15-must-do-steps-after-installing-drupal/</a></li>
+</ul>
+<br/>
+<ul>
+  <li><a href="http://www.devsaran.com/blog/6-drupal-seo-modules-improve-your-drupal-site-rankings">http://www.devsaran.com/blog/6-drupal-seo-modules-improve-your-drupal-site-rankings</a></li>
+  <li><a href="http://poplarware.com/articles/drupal_cheat_sheet#checklist">http://poplarware.com/articles/drupal_cheat_sheet#checklist</a>
+  </li>
+</ul>
+
+<h4>Credits</h4>
+<p>The Drupal Golive Checklist was created by <a href="http://dgo.to/@bechtold">Oskar Bechtold</a>, the CEO of <a
+    href="http://www.urlaubimglas.de">Urlaub im Glas</a> and long-time Drupal community member. Development was paid
+  for exclusively by Urlaub im Glas. Special thanks to <a href="http://dgo.to/@traviscarden">Travis Carden</a> who
+  created the <a href="http://dgo.to/checklistapi">Checklist API</a> and to <a href="http://dgo.to/@ben-finklea">Ben
+    Finklea</a> who created <a href="http://dgo.to/seo_checklist">Checklist API</a> which was mainly used as an
+  inspiration of the technical implementation of GOLIVE Checklist. </p>
+<p id="golive-checklist-intro-urlaubimglas">
+  <a href="http://www.urlaubimglas.de">{{ urlaubimglas_logo }}</a>
+</p>
