diff --git a/src/Controller/UserQueueShowList.php b/src/Controller/UserQueueShowList.php
index 8ed05cc..d13443d 100644
--- a/src/Controller/UserQueueShowList.php
+++ b/src/Controller/UserQueueShowList.php
@@ -111,7 +111,7 @@ class UserQueueShowList extends ControllerBase {
 	foreach ($queues as $q) {
         
      $account = \Drupal\user\Entity\User::load($q->uid); 
-     $name = $account->getUsername();
+     $name = $account->getDisplayName();
 
 	 $checking = $account->user_picture->first();
 
