diff --git a/src/Plugin/views/area/Link.php b/src/Plugin/views/area/Link.php
index a9b8d9b..714ab1e 100644
--- a/src/Plugin/views/area/Link.php
+++ b/src/Plugin/views/area/Link.php
@@ -263,7 +263,7 @@ class Link extends TokenizeAreaPluginBase {
       return [];
     }
 
-    $options = array(
+    $options = [
       'absolute' => !empty($this->options['absolute']) ? TRUE : FALSE,
       'alias' => FALSE,
       'entity' => NULL,
@@ -271,7 +271,7 @@ class Link extends TokenizeAreaPluginBase {
       'fragment' => NULL,
       'language' => NULL,
       'query' => [],
-    );
+    ];
 
     $path = $this->options['path'];
     if ($path != '<front>') {
diff --git a/tests/src/Kernel/Plugin/LinkTest.php b/tests/src/Kernel/Plugin/LinkTest.php
index 6ef6edf..83c9375 100644
--- a/tests/src/Kernel/Plugin/LinkTest.php
+++ b/tests/src/Kernel/Plugin/LinkTest.php
@@ -29,7 +29,7 @@ class LinkAreaTest extends ViewsKernelTestBase {
    *
    * @var array
    */
-  public static $testViews = array('test_entity_linkarea');
+  public static $testViews = ['test_entity_linkarea'];
 
   /**
    * {@inheritdoc}
@@ -40,7 +40,7 @@ class LinkAreaTest extends ViewsKernelTestBase {
       ViewTestData::createTestViews(get_class($this), ['views_test_linkarea']);
     }
     $random_label = $this->randomMachineName();
-    $data = array('bundle' => 'entity_test', 'name' => $random_label);
+    $data = ['bundle' => 'entity_test', 'name' => $random_label];
     $entity_test = $this->container->get('entity.manager')
       ->getStorage('entity_test')
       ->create($data);
