diff --git a/README.txt b/README.txt
index 33f2180..d8d89ec 100644
--- a/README.txt
+++ b/README.txt
@@ -12,7 +12,8 @@ CONTENTS OF THIS FILE
 INTRODUCTION
 ------------
 
-Counter field formatter is a simple module which animates number countups for a specific field (float/decimal/integer).
+Counter field formatter is a simple module which animates number countups for a
+specific field (float/decimal/integer).
 
  * For a full description of the module, visit the project page:
    https://www.drupal.org/project/counter_field_format
@@ -39,9 +40,11 @@ CONFIGURATION
 
  Instructions For Opening Image Fields In Popup:
 
-  * Navigate to Administration > Structure > Content types > Content Type > Manage
-    Display and change the format of the image field to "Counter Field Format".
-  * Click on the settings icon and provide the values for duration and easing style.
+  * Navigate to Administration > Structure > Content types > Content Type
+    > Manage Display and change the format of the image field to
+    "Counter Field Format".
+  * Click on the settings icon and provide the values for duration and
+    easing style.
 
 
 MAINTAINERS
diff --git a/counter_field_format.module b/counter_field_format.module
index 7641776..f096f8c 100644
--- a/counter_field_format.module
+++ b/counter_field_format.module
@@ -29,11 +29,10 @@ function counter_field_format_help($route_name, RouteMatchInterface $route_match
 function counter_field_format_theme() {
   return [
     'counter_field_format' => [
-      'variables' => array(
+      'variables' => [
         'item' => NULL,
         'counter_attributes' => NULL,
-      ),
+      ],
     ],
   ];
 }
-
diff --git a/src/Plugin/Field/FieldFormatter/CounterFieldFormatter.php b/src/Plugin/Field/FieldFormatter/CounterFieldFormatter.php
index e39241c..ceb9c57 100644
--- a/src/Plugin/Field/FieldFormatter/CounterFieldFormatter.php
+++ b/src/Plugin/Field/FieldFormatter/CounterFieldFormatter.php
@@ -46,7 +46,7 @@ class CounterFieldFormatter extends FormatterBase {
       '#type' => 'select',
       '#default_value' => $this->getSetting('image_style'),
       '#empty_option' => $this->t('None (original image)'),
-      '#options' => array('Swing', 'Linear'),
+      '#options' => ['Swing', 'Linear'],
       '#description' => $this->t('Animating style'),
     ];
     return $elements;
@@ -88,7 +88,7 @@ class CounterFieldFormatter extends FormatterBase {
         '#theme' => 'counter_field_format',
         '#item' => $item,
         '#counter_attributes' => $counter_attributes,
-        '#attached' => array('library' => array('counter_field_format/counter')),
+        '#attached' => ['library' => ['counter_field_format/counter']],
       ];
       // Make the user opted values available to js.
       $elements[$delta]['#attached']['drupalSettings']['counter_attributes']['duration'] = $duration;
