diff --git a/src/Form/WebformExtraFieldSettingsForm.php b/src/Form/WebformExtraFieldSettingsForm.php
index ca7597a..9c1c843 100644
--- a/src/Form/WebformExtraFieldSettingsForm.php
+++ b/src/Form/WebformExtraFieldSettingsForm.php
@@ -1,10 +1,5 @@
 <?php
 
-/**
- * @file
- * Contains WebformExtraFieldSettingsForm class.
- */
-
 namespace Drupal\webform_extra_field\Form;
 
 use Drupal\Core\Entity\ContentEntityType;
diff --git a/src/Plugin/ExtraField/Display/WebformExtraFieldDisplay.php b/src/Plugin/ExtraField/Display/WebformExtraFieldDisplay.php
index 5fd7b0d..6bf8bc8 100644
--- a/src/Plugin/ExtraField/Display/WebformExtraFieldDisplay.php
+++ b/src/Plugin/ExtraField/Display/WebformExtraFieldDisplay.php
@@ -1,10 +1,5 @@
 <?php
 
-/**
- * @file
- * Contains WebformExtraFieldDisplay.php.
- */
-
 namespace Drupal\webform_extra_field\Plugin\ExtraField\Display;
 
 use Drupal\Core\Entity\ContentEntityInterface;
@@ -12,7 +7,7 @@ use Drupal\extra_field_plus\Plugin\ExtraFieldPlusDisplayBase;
 use Drupal\webform\Entity\Webform;
 
 /**
- * Class WebformExtraFieldDisplay
+ * Class WebformExtraFieldDisplay.
  *
  * @ExtraFieldDisplay(
  *   id = "webform_extra_field_display",
@@ -38,7 +33,6 @@ class WebformExtraFieldDisplay extends ExtraFieldPlusDisplayBase {
     return [];
   }
 
-
   /**
    * {@inheritdoc}
    */
@@ -78,7 +72,7 @@ class WebformExtraFieldDisplay extends ExtraFieldPlusDisplayBase {
    * @return array
    *   The webform array option list.
    */
-  protected function getOptions(){
+  protected function getOptions() {
     $options = [];
     $webforms = Webform::loadMultiple();
     foreach ($webforms as $webform) {
