diff --git a/core/lib/Drupal/Core/Database/Query/Select.php b/core/lib/Drupal/Core/Database/Query/Select.php
index 3be5a2f..0fa13e7 100644
--- a/core/lib/Drupal/Core/Database/Query/Select.php
+++ b/core/lib/Drupal/Core/Database/Query/Select.php
@@ -776,7 +776,12 @@ public function __toString() {
     $comments = $this->connection->makeComment($this->comments);
 
     // SELECT
-    $query = $comments . 'SELECT ';
+    if($this->union) {
+      $query = $comments . '(SELECT ';
+    }
+    else {
+      $query = $comments . 'SELECT ';
+    }
     if ($this->distinct) {
       $query .= 'DISTINCT ';
     }
@@ -871,13 +876,16 @@ public function __toString() {
     // this query, but syntactically they all end up on the same level.
     if ($this->union) {
       foreach ($this->union as $union) {
-        $query .= ' ' . $union['type'] . ' ' . (string) $union['query'];
+        $query .= ")" . $union['type'] . ' (' . (string) $union['query'];
       }
     }
 
     if ($this->forUpdate) {
       $query .= ' FOR UPDATE';
     }
+    if($this->union) {
+      $query .= ")";
+    }
 
     return $query;
   }
