diff --git a/css/server_notice.css b/css/server_notice.css
index 7bd57b3..78ee49e 100644
--- a/css/server_notice.css
+++ b/css/server_notice.css
@@ -10,3 +10,11 @@
     z-index: 50;
     pointer-events: none;
 }
+
+.server-notice-border > .server-notice-notice {
+    position: fixed;
+    right: 0;
+    bottom: 0;
+    padding: 10px;
+    color: white;
+}
diff --git a/js/server_notice.js b/js/server_notice.js
index 6d31d8b..94c436a 100644
--- a/js/server_notice.js
+++ b/js/server_notice.js
@@ -2,11 +2,15 @@
 
     Drupal.behaviors.ServerNoticeBehavior = {
         attach: function (context, settings) {
-            console.log("in the js");
-            var server_notice_colour = drupalSettings.servernotice.colour;
-            var server_notice_border_element = $("<div class='server-notice-border'></div>")
-            $('body').after(server_notice_border_element);
-            $('.server-notice-border').css({ "border": "7px solid " + server_notice_colour });
+            var serverNoticeColour = drupalSettings.servernotice.colour;
+            var serverNoticeBorderElement = $("<div class='server-notice-border'></div>")
+            $('body').after(serverNoticeBorderElement);
+            $('.server-notice-border').css({ "border": "7px solid " + serverNoticeColour });
+            if (drupalSettings.servernotice.hasOwnProperty('notice')) {
+                var serverNoticeText = drupalSettings.servernotice.notice;
+                var serverNoticeNoticeElement = $("<div class='server-notice-notice'>" + serverNoticeText + "</div>").css({ "background": serverNoticeColour });
+                $(".server-notice-border").append(serverNoticeNoticeElement);
+            }
         }
     };
 
diff --git a/server_notice.module b/server_notice.module
index 69f36ea..3431901 100644
--- a/server_notice.module
+++ b/server_notice.module
@@ -9,7 +9,7 @@ use Drupal\server_notice\Entity\ServerNotice;
 
 function server_notice_page_attachments(array &$page) {
 
-    $host = \Drupal::request()->getHost();
+  $host = \Drupal::request()->getHost();
    $server_notice_entity_array = ServerNotice::loadMultiple(NULL);
 
     $active_server_notice = null;
@@ -24,5 +24,8 @@ function server_notice_page_attachments(array &$page) {
   if ($active_server_notice !== null) {
     $page['#attached']['library'][] = 'server_notice/server_notice';
     $page['#attached']['drupalSettings']['servernotice']['colour'] = $active_server_notice->get('colour')->value;
+    if (!$active_server_notice->get('notice')->isEmpty()) {
+        $page['#attached']['drupalSettings']['servernotice']['notice'] = $active_server_notice->get('notice')->value;
+    }
   }
 }
\ No newline at end of file
diff --git a/src/Entity/ServerNotice.php b/src/Entity/ServerNotice.php
index 618558b..6e50002 100644
--- a/src/Entity/ServerNotice.php
+++ b/src/Entity/ServerNotice.php
@@ -147,7 +147,7 @@ class ServerNotice extends ContentEntityBase {
       $fields['notice'] = BaseFieldDefinition::create('string')
       ->setLabel(t('Notice'))
       ->setDescription(t("Text to be displayed."))
-      ->setRequired(TRUE)
+      ->setRequired(FALSE)
       ->setTranslatable(FALSE)
       ->setDisplayOptions('form', array(
         'weight' => -5,
