diff --git a/src/Plugin/Block/countdown.php b/src/Plugin/Block/countdown.php
index fd14d89..7f3ff99 100755
--- a/src/Plugin/Block/countdown.php
+++ b/src/Plugin/Block/countdown.php
@@ -30,13 +30,13 @@ class Countdown extends BlockBase {
 
     if ($url != "") {
       if (UrlHelper::isExternal($url)) {
-        $event_name = Link::fromTextAndUrl($event_name, Url::fromUri($url, array()))->toString();
+        $event_name = Link::fromTextAndUrl($event_name, Url::fromUri($url, []))->toString();
       }
       else {
         if ($url == "<front>") {
           $url = "/";
         }
-        $event_name = Link::fromTextAndUrl($event_name, Url::fromUri('internal:/' . ltrim($url, '/'), array()))->toString();
+        $event_name = Link::fromTextAndUrl($event_name, Url::fromUri('internal:/' . ltrim($url, '/'), []))->toString();
       }
     }
     $accuracy = isset($config['accuracy']) ? $config['accuracy'] : '';
