diff --git a/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php b/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php
index a33e72e..2eced41 100644
--- a/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php
+++ b/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php
@@ -8,6 +8,7 @@ use Drupal\Core\Entity\Element\EntityAutocomplete;
 use Drupal\Core\Field\FieldItemListInterface;
 use Drupal\Core\Field\WidgetBase;
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\entityreference_view_mode\Plugin\Field\FieldType\EntityReferenceViewModeFieldType;
 
 
 /**
@@ -113,6 +114,11 @@ class EntityReferenceViewModeFieldWidget extends WidgetBase {
 
     $bundles = array_filter($selected[$target_type]['bundles']);
 
+    // If no bundles have been selected then return them all!
+    if (empty($bundles)) {
+      $bundles = EntityReferenceViewModeFieldType::availableBundles($target_type);
+    }
+
     // Generic alter function.
     \Drupal::moduleHandler()
       ->alter('entityreference_view_mode_bundles', $bundles, $target_type);
diff --git a/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidgetTrait.php b/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidgetTrait.php
index 7cd354d..10f228e 100644
--- a/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidgetTrait.php
+++ b/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidgetTrait.php
@@ -58,6 +58,9 @@ trait EntityReferenceViewModeFieldWidgetTrait {
       if ($entity) {
         $selections['content'] = $entity;
         $selections['bundle'] =  $entity->bundle();
+      } else {
+          $selections['content'] = '';
+          $selections['bundle'] = '';
       }
     }
 
