diff --git a/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php b/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php
index a33e72e..7bb1b9b 100644
--- a/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php
+++ b/src/Plugin/Field/FieldWidget/EntityReferenceViewModeFieldWidget.php
@@ -8,6 +8,7 @@ use Drupal\Core\Entity\Element\EntityAutocomplete;
 use Drupal\Core\Field\FieldItemListInterface;
 use Drupal\Core\Field\WidgetBase;
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\entityreference_view_mode\Plugin\Field\FieldType\EntityReferenceViewModeFieldType;
 
 
 /**
@@ -87,7 +88,12 @@ class EntityReferenceViewModeFieldWidget extends WidgetBase {
       $view_modes[$target_type . '.' . $key] = $all_view_modes[$target_type . '.' . $key]->label();
     }
 
-    // Generic alter function.
+    // If no view modes have been selected then return them all!
+    if (empty($view_modes)) {
+        $view_modes = EntityReferenceViewModeFieldType::availableViewModes($target_type);
+    }
+      
+      // Generic alter function.
     \Drupal::moduleHandler()
       ->alter('entityreference_view_mode_modes', $view_modes, $target_type, $bundle);
 
@@ -113,6 +119,11 @@ class EntityReferenceViewModeFieldWidget extends WidgetBase {
 
     $bundles = array_filter($selected[$target_type]['bundles']);
 
+    // If no bundles have been selected then return them all!
+    if (empty($bundles)) {
+        $bundles = EntityReferenceViewModeFieldType::availableBundles($target_type);
+    }
+
     // Generic alter function.
     \Drupal::moduleHandler()
       ->alter('entityreference_view_mode_bundles', $bundles, $target_type);
