diff --git b/src/Form/PaymentMethodForm.php a/src/Form/PaymentMethodForm.php
index 4e038fb..72f2a46 100644
--- b/src/Form/PaymentMethodForm.php
+++ a/src/Form/PaymentMethodForm.php
@@ -82,45 +82,18 @@ class PaymentMethodForm extends FormBase {
 
     $form['#attributes']['id'] = 'payment-method-form';
 
-    // The form submit handler isn't triggered by jQuery's $().submit().
-    // This hidden button is added for jQuery to click on to submit the form.
-    $form['final_submit'] = [
-      '#type' => 'submit',
-      '#name' => 'final_submit',
-      '#attributes' => [
-        'id' => 'final-submit',
-        'class' => [
-          'visually-hidden',
-        ],
-      ],
-      '#submit' => [[$this, 'submitForm']],
-    ];
-
     $form['uid'] = [
       '#type' => 'value',
       '#value' => $user->id(),
     ];
 
-    $form['dropin_container'] = [
-      '#type' => 'html_tag',
-      '#tag' => 'div',
-      '#attributes' => [
-        'id' => 'dropin-container',
-      ],
-    ];
-
-    $form['#attached']['library'][] = 'braintree_cashier/dropin_support';
-    $form['#attached']['drupalSettings']['braintree_cashier'] = [
-      'authorization' => $this->billableUser->generateClientToken($user),
-      'acceptPaypal' => (bool) $this->config('braintree_cashier.settings')->get('accept_paypal'),
-    ];
+    $form['dropin_ui'] = $this->billableUser->getDropinUiFormElement($user);
 
     $form['actions'] = [
       '#type' => 'actions',
       '#button_type' => 'primary',
       'submit' => [
         '#type' => 'submit',
-        '#disabled' => TRUE,
         '#attributes' => [
           'id' => 'submit-button',
           'class' => [
@@ -140,7 +113,7 @@ class PaymentMethodForm extends FormBase {
     $form['payment_method_nonce'] = [
       '#type' => 'hidden',
       '#attributes' => [
-        'id' => 'payment-method-nonce',
+        'id' => 'payment_method_nonce',
       ],
     ];
 
diff --git b/src/Form/SignupForm.php a/src/Form/SignupForm.php
index 3f012bb..9d3c708 100644
--- b/src/Form/SignupForm.php
+++ a/src/Form/SignupForm.php
@@ -128,9 +128,9 @@ class SignupForm extends PlanSelectFormBase {
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
 
-    // This hidden button will be the triggering element since it's first. This
-    // is needed to distinguish between clicks on the "Confirm Coupon" button,
-    // and the "Sign up" button.
+    // When the form is submitted by the Braintree JS, this hidden button will
+    // be the triggering element since it's first. This is needed to distinguish
+    // between clicks on the "Confirm Coupon" button, and the "Sign up" button.
     $form['final_submit'] = [
       '#type' => 'submit',
       '#name' => 'final_submit',
@@ -149,24 +149,11 @@ class SignupForm extends PlanSelectFormBase {
       $user = $this->userStorage->load($this->currentUser->id());
     }
 
-    $form['dropin_container'] = [
-      '#type' => 'html_tag',
-      '#tag' => 'div',
-      '#attributes' => [
-        'id' => 'dropin-container',
-      ],
-    ];
-
-    $form['#attached']['library'][] = 'braintree_cashier/dropin_support';
-    $form['#attached']['drupalSettings']['braintree_cashier'] = [
-      'authorization' => $this->billableUser->generateClientToken($user),
-      'acceptPaypal' => (bool) $this->config('braintree_cashier.settings')->get('accept_paypal'),
-    ];
+    $form['dropin_ui'] = $this->billableUser->getDropinUiFormElement($user);
 
     $form['submit'] = [
       '#type' => 'submit',
       '#name' => 'submit_signup',
-      '#disabled' => TRUE,
       '#attributes' => [
         'id' => 'submit-button',
       ],
