diff --git a/src/Entity/WebhookConfig.php b/src/Entity/WebhookConfig.php
index 2e28514..659d808 100644
--- a/src/Entity/WebhookConfig.php
+++ b/src/Entity/WebhookConfig.php
@@ -243,7 +243,9 @@ class WebhookConfig extends ConfigEntityBase implements WebhookConfigInterface {
    */
   public function preSave(EntityStorageInterface $storage) {
     parent::preSave($storage);
-    $this->events = serialize($this->events);
+    $this->events = serialize(
+      array_filter($this->events)
+    );
   }
 
 }
