diff --git a/src/Plugin/ad_entity/AdType/ViType.php b/src/Plugin/ad_entity/AdType/ViType.php
index db2508e..31d4d54 100644
--- a/src/Plugin/ad_entity/AdType/ViType.php
+++ b/src/Plugin/ad_entity/AdType/ViType.php
@@ -136,7 +136,7 @@ class ViType extends AdTypeBase {
       '#title' => $this->stringTranslation->translate('Font size'),
       '#default_value' => !empty($settings['font_size']) ? $settings['font_size'] : '',
       '#size' => 15,
-      '#description' => $this->stringTranslation->translate("Font size for the video title showing in the native video unit. E.g. '14'."),
+      '#description' => $this->stringTranslation->translate("Font size for the video title showing in the native video unit. E.g. '14px'."),
     ];
 
     return $element;
